rotkehlchen/api/rest.py

```diff
@@ -3474,9 +3474,20 @@ def get_history_events(
                 action_type=ActionType.HISTORY_EVENT,
             )
 
+        db_accounting = DBAccountingRules(self.rotkehlchen.data.db)
+        if group_by_event_ids is True:
+            valid_entry_combinations = db_accounting.get_valid_entry_combinations([event for _, event in events_result])  # type: ignore # noqa: E501
+        else:
+            valid_entry_combinations = db_accounting.get_valid_entry_combinations(events_result)  # type: ignore
+
+        def has_accounting_rule(entry: 'HistoryBaseEntry') -> dict[str, bool]:
```

Comment:
why do it like this? Why not edit the `serialize_for_api()` function which is the function made for this?

Along with the return of the modified function I asked for in the comment on `get_valid_entry_combination` it should be straightforward and easier.